react/next/index.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { FC, useReducer, useEffect, useRef } from 'react'
+
+import { SliderProps } from './typings'
+import {
+  getInitialState,
+  throwError,
+  getItemClientSideWidth,
+  populateNextSlides,
+  populatePreviousSlides,
+} from './utils/index'
+import Dots from './Dots'
+import Arrow from './Arrow'
+import Slides from './Slides'
+import SliderTrack from './SliderTrack'
+import reducer from './stateReducer'
+
+/**
+ * Slider's main component
+ */
+const SliderNext: FC<SliderProps> = props => {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [state, dispatch] = useReducer(reducer, {
+    itemWidth: 0,
+    slidesToShow: 0,
+    currentSlide: 0,
+    totalItems: React.Children.count(props.children),
+    deviceType: '',
+    domLoaded: false,
+    transform: 0,
+    containerWidth: 0,
+  })
+
+  useEffect(() => {
+    const setNewState = (shouldCorrectItemPosition: boolean) => {
+      const { responsive } = props
+      Object.keys(responsive).forEach(item => {
+        const { breakpoint, items } = responsive[item]
+        const { max, min } = breakpoint
+        if (window.innerWidth >= min && window.innerWidth <= max) {
+          if (containerRef && containerRef.current) {
+            const containerWidth = containerRef.current.offsetWidth
+            const itemWidth: number = getItemClientSideWidth(
+              items,
+              containerWidth
+            )
+            dispatch({
+              type: 'loadAndCorrect',
+              payload: {
+                slidesToShow: items,
+                deviceType: item,
+                containerWidth,
+                itemWidth,
+                shouldCorrectItemPosition,
+              },
+            })
+          } else {
+            dispatch({
+              type: 'load',
+              payload: {
+                slidesToShow: items,
+                deviceType: item,
+              },
+            })
+          }
+        }
+      })
+    }
+
+    const onResize = (value?: any): void => {
+      setNewState(!(value && !props.infinite))
+    }
+
+    setNewState(false)
+    window.addEventListener('resize', onResize)
+    return () => window.removeEventListener('resize', onResize)
+  }, [])
```

Comment:
Is this right? A huge function that depends on a lot of stuff with a clear dep array?

Reply:
If `responsive` change, for instance? Shouldn't we _react_ to that?

Reply:
I really cannot see a use case may cause `responsive` prop to change values. 🤔